test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,154 @@
+                        file = File.createTempFile("prefix", ".suffix");
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    } finally {
+                        if (file != null && file.exists())
+                            if (!file.delete())
+                                throw new RuntimeException(file + " not deleted");
+                    }
+                }
+                case "nio" -> {
+                Path path = null;
+                    try {
+                        path = Files.createTempFile("prefix", ".suffix");
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    } finally {
+                        if (path != null)
+                            if (!Files.deleteIfExists(path))
+                                throw new RuntimeException(path + " not deleted");
+                    }
                 }
-            } else if (arg.equals("nio")) {
-                try {
-                    Files.createTempFile("prefix", ".suffix");
-                } catch (Exception e) {
-                    e.printStackTrace();
+                default -> {
+                    throw new RuntimeException("unknown case: " + arg);
                 }
-            } else {
-                throw new Exception("unknown case: " + arg);
             }
         }
+    }
 
-        if (args.length == 0) {
-            // standard test with default setting for java.io.tmpdir
-            testMessageNotExist(0, ioWarningMsg, "TempDirDoesNotExist", "io");
-            testMessageNotExist(0, ioWarningMsg, "TempDirDoesNotExist", "nio");
-
-            // valid custom java.io.tmpdir
-            testMessageNotExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + userDir,
-                    "TempDirDoesNotExist", "io");
-            testMessageNotExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + userDir,
-                    "TempDirDoesNotExist", "nio");
-
-            // invalid custom java.io.tmpdir
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io");
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "nio");
+    private static String tempDir() {
+        String timeStamp = String.valueOf(System.currentTimeMillis());
+        return Path.of(USER_DIR, "non-existing-", timeStamp).toString();
+    }
 
-            // test with security manager
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "io");
+    public static Stream<Arguments> tempDirSource() {
+        return Stream.of(Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" +
+                                          tempDir(),
```

Comment:
These lines could be joined. Or perhaps move the "+" to the beginning of the next line as in the others below.

Reply:
Fixed in b3ac8c7b1242281bda86f7dc98efccf2bcefc7f6.